bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -0,0 +1,307 @@
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.openthermgateway.handler;
+
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.openthermgateway.OpenThermGatewayBindingConstants;
+import org.openhab.binding.openthermgateway.internal.DataItem;
+import org.openhab.binding.openthermgateway.internal.DataItemGroup;
+import org.openhab.binding.openthermgateway.internal.GatewayCommand;
+import org.openhab.binding.openthermgateway.internal.GatewayCommandCode;
+import org.openhab.binding.openthermgateway.internal.LogLevel;
+import org.openhab.binding.openthermgateway.internal.Message;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayCallback;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayConfiguration;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayConnector;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewaySocketConnector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OpenThermGatewayHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Arjen Korevaar - Initial contribution
+ */
+@NonNullByDefault
+public class OpenThermGatewayHandler extends BaseThingHandler implements OpenThermGatewayCallback {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenThermGatewayHandler.class);
+
+    @Nullable
+    private OpenThermGatewayConfiguration config;
+
+    @Nullable
+    private OpenThermGatewayConnector connector;
+
+    public OpenThermGatewayHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing OpenTherm Gateway handler for uid '{}'", getThing().getUID());
+
+        updateStatus(ThingStatus.OFFLINE);
+
+        config = getConfigAs(OpenThermGatewayConfiguration.class);
+
+        connect();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Received channel: {}, command: {}", channelUID, command);
+
+        try {
+            if (command.toFullString() != "REFRESH") {
```

Comment:
Don't compare string with `!=`. But you should check here on instanceof 
```suggestion
            if (!command instanceof RefreshType) {
```

You do need to handle the `RefreshType` command  by updating the channel with the known (or unknown) value.

Reply:
Changed comparison. Not sure how to respond to refreshtype commands. The binding doesn't keep track of the values, nor can it read the values from the OTGW device. The device reports the values continuously to the socket and the binding maps the incoming values to the channels.

Reply:
Hi @Hilbrand, could you please review this PR and this comment in particular ?